### PR TITLE
Eman/database query

### PR DESCRIPTION
**Pull request by emanNahed**

### CODE REVIEW

**Review by NourAdawi — DISMISSED** [review]

**Review by divluffy — APPROVED** [review]

**Review by divluffy — DISMISSED** [review]

**Review by asmaaamin — DISMISSED** [review]

File: server/DB/config/connections.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const {pool, Pool} = require('pg');
```
Comment (Mu7ammadAbed) [inline server/DB/config/connections.js:1]:
```suggestion
const {Pool} = require('pg');
```

File: server/DB/config/connections.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const options = {
+    host: params.hostname,
+    port: params.port,
+    database: params.pathname.split('/')[1],
+    max: process.env.DB_MAX_CONNECTIONS || 2,
+    user: params.username,
+    password: params.password,
+    ssl: params.hostname !== 'localhost'
+}
```
Comment (Mu7ammadAbed) [inline server/DB/config/connections.js:20]: This need to be replaced with `connectionString` and edit the `ssl`

File: server/DB/query/email-exist.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const connection = require('../config/connections');
+
+const emailExists= (email) => {
+    const sql = {
+        text: 'SELECT * FROM users WHERE email = $1',
```
Comment (Mu7ammadAbed) [inline server/DB/query/email-exist.js:5]: There is no need to select all the data from the table just to check if the user exists or no
returning the id maybe is more than enough

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: server/DB/config/connections.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+const {pool, Pool} = require('pg');
+require('env2')('./config.env');
```
Comment (Mu7ammadAbed) [inline server/DB/config/connections.js:2]: I think you are going to use the environment variables in other places than the connection file like in auth controllers for `SECRET_KEY, so you should require the `env` in the `app.js` maybe

File: server/DB/config/connections.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  
+const DB_URL = process.env.DB_URL;
```
Comment (Mu7ammadAbed) [inline server/DB/config/connections.js:4]:
```suggestion
const {DB_URL} = process.env;
```

File: server/DB/config/connections.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    ssl: params.hostname !== 'localhost'
+}
+
+
+module.exports= new Pool(options);
+
+
+
+
+
```
Comment (Mu7ammadAbed) [inline server/DB/config/connections.js:28]:
```suggestion

```

File: server/DB/query/email-exist.js (hunk 3 above)
Comment (emanNahed) [reply server/DB/query/email-exist.js]: I need to check the validate of the password with the same method, then pass data via cookie

that's why I select all of them

**Pull request merged**